packages/esm-patient-chart-app/src/visit/visit-form/visit-form.component.tsx

```diff
@@ -244,6 +245,72 @@ const StartVisitForm: React.FC<StartVisitFormProps> = ({
     return validSubmission;
   }, [setError]);
 
+  const handleVisitAttributes = useCallback(
+    async (visitAttributes: { [p: string]: string }, visitUuid: string) => {
+      const existingAttributes = visitToEdit?.attributes?.map((attribute) => attribute.attributeType.uuid) || [];
+      const attributes = Object.entries(visitAttributes).map(([key, value]) => ({
+        attributeType: key,
+        value: value as string,
+      }));
+
+      for (const attribute of attributes) {
+        if (attribute.attributeType && existingAttributes.includes(attribute.attributeType)) {
+          const attributeToEdit = visitToEdit.attributes.filter(
+            (attr) => attr.attributeType.uuid === attribute.attributeType,
+          )[0];
+
+          if (attributeToEdit) {
+            // continue to next attribute if the previous value is same as new value
+            if (attributeToEdit.value.display === attribute.value) continue;
+            if (attribute.value) {
+              // Update attribute with updated value
+              await openmrsFetch(`/ws/rest/v1/visit/${visitUuid}/attribute/${attributeToEdit.uuid}`, {
+                method: 'POST',
+                headers: { 'Content-type': 'application/json' },
+                body: { value: attribute.value },
+              }).catch((err) => {
+                showSnackbar({
+                  title: t('errorUpdatingVisitAttribute', 'Error on updating visit attribute'),
```

Comment:
Ideally, we'd also have the name of the attribute we're updating here.

```suggestion
                  title: t('errorUpdatingVisitAttribute', 'Could not update visit attribute'),
```